nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -385,8 +385,8 @@ in
       };
 
       default = mkOption {
-        default = 0;
-        type = types.int;
+        default = "0";
+        type = types.str;
```

Comment:
How about?

```
type = types.either types.int types.str;
apply = toString;
``

Reply:
oh neat. didnt know you could do the either thing